### PR TITLE
Pre-commit hooks

### PR DESCRIPTION
**Pull request by jpopelka**

### CODE REVIEW

File: content/posts/pre-commit-hooks.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+---
+title: "Pre-commit Hooks"
+date: 2023-05-10T17:00:00
+weight: 90
+---
+
+## pre-commit
+
+[pre-commit](https://pre-commit.com) is a wonderful tool that saves you
+a lot of time by outomatically checking your changes before you
```
Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:10]:
```suggestion
a lot of time by automatically checking your changes before you
```

File: content/posts/pre-commit-hooks.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+          - id: check-rebase
+            args: [upstream_url]
+
+### validate-config
+
+Packit uses a [YAML configuration file](https://packit.dev/docs/configuration)
+in an upstream repository.
+We have a [packit validate-config](https://packit.dev/docs/cli/validate-config) command
+to check it, but it's easy to forget (to run it) and realize a typo after you've
+commited and pushed the changes and waited for some time for Packit to tell you
```
Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:46]:
```suggestion
committed and pushed the changes and waited for some time for Packit to tell you
```

File: content/posts/pre-commit-hooks.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+that in a PR.
+
+It's much faster to catch the problem before committing and/or pushing the changes.
+
+#### validate-config-in-container
+
+This hook runs (only if there's been a change in the `.packit.yaml`)
+packit in a container (from [our image](https://quay.io/repository/packit/packit)),
```
Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:54]:
```suggestion
`packit` in a container (from [our image](https://quay.io/repository/packit/packit)),
```

Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:54]:
```suggestion
`packit` in a container (from [our image](https://quay.io/repository/packit/packit)),
```

File: content/posts/pre-commit-hooks.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+        hooks:
+          - id: validate-config
+
+##### But why the hook doesn't install `packit` itself?
+
+Right, typically, when you run a pre-commit hook for the first time,
+it installs everything it needs.
+Like in case of Python, pre-commit `pip` installs all the dependencies.
+
+The problem in our case is that packit has a lots of dependencies and
```
Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:82]:
```suggestion
The problem in our case is that `packit` has a lot of dependencies and
```

File: content/posts/pre-commit-hooks.md (hunk 3 above)
Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:54]:
```suggestion
`packit` in a container (from [our image](https://quay.io/repository/packit/packit)),
```

File: content/posts/pre-commit-hooks.md (hunk 4 above)
Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:82]:
```suggestion
The problem in our case is that `packit` has a lot of dependencies and
```

**Review by lbarcziova — COMMENTED** [review]
nicely written!!

**Review by lachmanfrantisek — APPROVED** [review]
Well written!

I really like describing other uses of pre-commit as well.

File: content/posts/pre-commit-hooks.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+        hooks:
+          - id: check-rebase
+            args: [upstream_url]
+
+### validate-config
+
+Packit uses a [YAML configuration file](https://packit.dev/docs/configuration)
+in an upstream repository.
+We have a [packit validate-config](https://packit.dev/docs/cli/validate-config) command
+to check it, but it's easy to forget (to run it) and realize a typo after you've
```
Comment (lbarcziova) [inline content/posts/pre-commit-hooks.md:45]:
```suggestion
to check it, but it's easy to forget (to run it) and notice a typo after you've
```

**Review by lbarcziova — APPROVED** [review]
:clap:

**Pull request merged**